pkg/sds/provider/ca_provider.go

```diff
@@ -1,4 +1,4 @@
-package ca
+package provider
```

Comment:
I think `ca` and `identity` packages were more meaningful because `provider` as a package name doesn't tell you much. I'd leave packages `ca` and `identity` and rename interfaces and funcs: 
- `provider.CaProvider` -> `ca.Provider`
- `provider.NewCaProvider` -> `ca.NewProvider`
- `provider.IdentityCertProvider` -> `identity.Provider`
- `provider.NewIdentityProvider` -> `identity.NewProvider`

And probably I'd delete directory `pkg/sds/provider` to have:

- `/pkg/sds`
    - `ca`
        - `provider.go` 
    - `identity`
        - `provider.go` 
    - `server`
        - `...`

What do you think?